lib/streamlit/elements/write.py

```diff
@@ -181,6 +181,9 @@ def flush_buffer():
             # Order matters!
             if isinstance(arg, str):
                 string_buffer.append(arg)
+            elif type_util.is_snowpark_dataframe(arg):
+                flush_buffer()
+                self.dg.dataframe(arg)
```

Comment:
This also needs a test. It can be simple - all we care is that `self.dg.dataframe` is called if the `is_snowpark_dataframe` predicate is true, so you can mock `is_snowpark_dataframe` and assert that `self.dg.dataframe` is called when it returns true.

Reply:
Thanks for the review, the tests for this func are present now, please see my comment above 👍